openvdb_houdini/houdini/SOP_OpenVDB_From_Particles.cc

```diff
@@ -1128,6 +1127,17 @@ VDB_NODE_OR_CACHE(VDB_COMPILABLE_SOP, SOP_OpenVDB_From_Particles)::cookVDBSop(OP
              return error();
         }
 
+        bool needLeveLSet = (outputLevelSetGrid || outputFogVolumeGrid || outputBoundingMaskGrid);
+
+        // Backwards compatibility for when attribute grids are being requested
+        // but no level set output is desired. outputInteriorMaskGrid will always
+        // be false for older nodes. We need to provide convertWithAttributes()
+        // a valid level set in this case.
+
+        if (outputAttributeGrid && !outputLevelSetGrid && !outputInteriorMaskGrid) {
+            needLeveLSet = true;
+        }
```

Comment:
@pcucka Feel free to move this back up to be part of the assignment to needLeveLSet - split it out to be clear but perhaps this is always the intended behaviour anyway.

Reply:
Changing line 1122 to
```
needLevelSet = (outputLevelSetGrid || outputFogVolumeGrid
    || outputBoundingMaskGrid || outputAttributeGrid);
```
would be preferable.  Is that what you meant?

Reply:
Yeah exactly - though you can probably do:

```
needLevelSet = (outputLevelSetGrid || outputFogVolumeGrid
    || outputBoundingMaskGrid || (outputAttributeGrid && !outputInteriorMaskGrid));
```
to avoid building the surface if the mask is selected.

Reply:
That sounds good.  Do you want to update your PR?

Reply:
sure, done